source/channel_bootstrap.c

```diff
@@ -1402,12 +1606,20 @@ void s_on_server_connection_result(
 
 error_cleanup:
     /* no channel is created */
-    connection_args->incoming_callback(connection_args->bootstrap, aws_last_error(), NULL, connection_args->user_data);
-
+    (void)socket; // to avoid expression error after a label
     struct aws_allocator *allocator = new_socket->allocator;
+
+    struct socket_shutdown_server_connection_result_args *close_args =
```

Comment:
I see this pattern everywhere.  Can we create some helper functions and use them instead?